File: solutions/leetcode/longest-subarray-of-1s-after-deleting-one-element/main.go
func longestSubarray(nums []int) int {
	result := 0
	deleted := 0
	l := 0
	for r, v := range nums {
		if v == 0 {
			deleted++
		}
		for deleted > 1 {
			if nums[l] == 0 {
				deleted--
			}
			l++
		}
		result = max(result, r-l)
	}
	return result
}

File: solutions/leetcode/longest-subarray-of-1s-after-deleting-one-element/main.py
class Solution:

    def longestSubarray(self, nums: List[int]) -> int:
        result = 0
        deleted = 0
        l = 0
        for r in range(len(nums)):
            if nums[r] == 0:
                deleted += 1
            while deleted > 1:
                if nums[l] == 0:
                    deleted -= 1
                l += 1
            result = max(result, r - l)
        return result
